go/runtime/registry/registry.go

```diff
@@ -284,21 +319,31 @@ func (r *runtimeRegistry) addSupportedRuntime(ctx context.Context, id common.Nam
 	var ns common.Namespace
 	copy(ns[:], id[:])
 
-	storageBackend, err := storage.New(ctx, path, ns, r.identity,
-		storage.WithConsensus(r.consensus),
-		storage.WithRuntime(rt),
-	)
+	//	impl, err = client.New(ctx, namespace, identity, o.consensus.Scheduler(), o.consensus.Registry(), o.runtime)
+	//storageBackend, err := storage.NewLocalBackend(ctx, path, ns, r.identity,
+	//	storage.WithConsensus(r.consensus),
+	//	storage.WithRuntime(rt),
+	//)
+	/*storageBackend, err := client.New(ctx, ns, r.identity, r.consensus.Scheduler(), r.consensus.Registry(), rt)
 	if err != nil {
 		return fmt.Errorf("runtime/registry: cannot create storage for runtime %s: %w", id, err)
 	}
 	defer func() {
 		if rerr != nil {
 			storageBackend.Cleanup()
 		}
-	}()
+	}()*/
+	storageWaiter := func(cancelCh chan struct{}) storageAPI.Backend {
+		select {
+		case <-cancelCh:
+			return nil
+		case <-r.storageSetCh:
+			return rt.storage
+		}
+	}
 
 	// Create runtime tag indexer.
-	tagIndexer, err := tagindexer.New(path, cfg.TagIndexer, history, r.consensus.RootHash(), storageBackend)
+	tagIndexer, err := tagindexer.New(path, cfg.TagIndexer, history, r.consensus.RootHash(), storageWaiter)
```

Comment:
Instead of using the storage waiter, could you just defer starting the tag indexer until storage is fully initialized and then pass the storage backend to the start method? I think this is the only use case for `WaitForBackend` and so it could be removed.